localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py

```diff
@@ -61,22 +73,57 @@ def __call__(
             response_parameters=response_parameters,
         )
 
-        # we can also apply the Response Templates, they will depend on status code + accept header?
-
-        # then responseOverride
+        # We then fetch a response templates and apply the template mapping
+        response_template = self.get_response_template(
+            integration_response=integration_response, request=context.invocation_request
+        )
+        body, response_override = self.render_request_template_mapping(
+            context=context, template=response_template, response=response
+        )
 
         # here we update the `Response`. We basically need to remove all headers and replace them with the mapping, then
         # override them if there are overrides.
-        # for the body, it will maybe depend on configuration?
-        for key in list(response.headers.keys(lower=True)):
-            if key not in ("connection", "content-type"):
-                del response.headers[key]
+        # The status code is pretty straight forward. By default, it would be set by the integration response,
+        # unless there was an override
+        response.status_code = int(integration_response["statusCode"])
+        if response_status_override := response_override["status"]:
+            # maybe make a better error message format, same for the overrides for request too
+            LOG.debug("Overriding response status code: '%s'", response_status_override)
+            response.status_code = response_status_override
+
+        # Create a new headers object that we can manipulate before overriding the original response headers
+        headers = Headers(response_data_mapping.get("header"))
+        if header_override := response_override["header"]:
+            LOG.debug("Response header overrides: %s", header_override)
+            headers.update(header_override)
+
+        # When trying to override certain headers, they will instead be remapped
+        # There may be other headers, but these have been confirmed on aws
+        remapped_headers = (
+            "connection",
+            "content-length",
+            "date",
+            "x-amzn-requestid",
+            "content-type",
+        )
+        for header in remapped_headers:
+            if value := headers.get(header):
+                headers[f"x-amzn-remapped-{header}"] = value
+                headers.remove(header)
```

Comment:
This is a really great find! It's also very similar to what is done in the `HTTP_PROXY` integration.
[The big table in the documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-known-issues.html#api-gateway-known-issues-rest-apis) shares more insights, and it seems it can be different from each integration... this is painful. 

I think we should tackle this in a follow up, I wonder how we can generalize this. Maybe a util? Because I was thinking of a handler, but in the integration response step, this is very specific, it happens after mapping/overriding but before passing through, unlike the proxy where it just happens on the response directly. This is going to be fun to test 😅 and we also need to do it for requests, it seems a few are dropped, nothing remapped.